irc/handlers.go

```diff
@@ -2270,6 +2279,53 @@ func rehashHandler(server *Server, client *Client, msg ircmsg.IrcMessage, rb *Re
 	return false
 }
 
+// RELAYMSG <channel> <spoofed nick> :<message>
+func relaymsgHandler(server *Server, client *Client, msg ircmsg.IrcMessage, rb *ResponseBuffer) (result bool) {
+	channel := server.channels.Get(msg.Params[0])
+	if channel == nil {
+		rb.Add(nil, server.name, ERR_NOSUCHCHANNEL, client.Nick(), utils.SafeErrorParam(msg.Params[0]), client.t("No such channel"))
+		return false
+	}
+
+	if !(channel.ClientIsAtLeast(client, modes.ChannelOperator) || client.HasRoleCapabs("relaymsg-anywhere")) {
+		rb.Add(nil, server.name, "FAIL", "RELAYMSG", "NOT_PRIVED", client.t("Only channel operators or ircops with the 'relaymsg-anywhere' role can relay messages"))
+		return false
+	}
+
+	rawMessage := msg.Params[2]
+	if strings.TrimSpace(rawMessage) == "" {
+		rb.Add(nil, server.name, "FAIL", "RELAYMSG", "BLANK_MSG", client.t("The message must not be blank"))
+		return false
+	}
+	message := utils.MakeMessage(rawMessage)
+
+	nick := msg.Params[1]
+	_, err := CasefoldName(nick)
+	if err != nil {
+		rb.Add(nil, server.name, "FAIL", "RELAYMSG", "INVALID_NICK", client.t("Invalid nickname"))
+		return false
+	}
+	if !strings.Contains(nick, "/") {
+		rb.Add(nil, server.name, "FAIL", "RELAYMSG", "INVALID_NICK", fmt.Sprintf(client.t("Relayed nicknames MUST contain the relaymsg separator %s"), "/"))
+		return false
+	}
+
+	//TODO: add to history here??
```

Comment:
I'd definitely prefer if we did this. As edk0 put it: "in the world we are, or [at] least should be, moving towards, storage and transmission are equivalent".

Should be as easy as calling `channel.AddHistoryItem` with the appropriate args.

Reply:
Aha I think that's mostly me crying out for help with how to do history. But awesome I'll look at that function

Reply:
Shooould be done now